lightning/src/ln/onion_utils.rs

```diff
@@ -1830,13 +1834,13 @@ where
 					}),
 					Ok((_, None)) => Err(OnionDecodeErr::Malformed {
 						err_msg: "Non-final Trampoline onion data provided to us as last hop",
-						// todo: find more suitable error code
-						err_code: 0x4000 | 22,
+						err_code: INVALID_ONION_BLINDING,
+						trampoline_onion_blinding: true,
 					}),
 					Ok((_, Some(_))) => Err(OnionDecodeErr::Malformed {
 						err_msg: "Final Trampoline onion data provided to us as intermediate hop",
-						// todo: find more suitable error code
-						err_code: 0x4000 | 22,
+						err_code: INVALID_ONION_BLINDING,
+						trampoline_onion_blinding: true,
 					}),
```

Comment:
Somewhat annoying, but if we want to be accurate to the spec I believe we shouldn't be returning `INVALID_ONION_BLINDING` for non-blinded trampoline forward/receives, or for blinded receives where the intro blinding point is set. Apologies if I misunderstood some context on this when we discussed offline. I think if we matched on the specific variants instead of `_` this change should be doable?